### DISCUSSION TITLE
Generic Hydrostat vs Generic Thermostat

### DISCUSSION THREAD

**Discussion by sharoltd** [home-assistant/architecture#706] [Entity Models] [answered]
Hi! As I understood right Generic Hydrostat is a almost complete analogue of the Generic Thermostate? Why not match the look of theese two devices? The appearance of the hydrostat is not quite  functional at the moment. It would be nice to add "on-off" icons directly to the card itself, add the current humidity from the sensor and so on, as already implemented in Generic Thermostate. I think it's very easy for developers. What do you think?
Thanks!

**Comment by Shulyaka** [comment, marked as answer]
I've raised https://github.com/home-assistant/frontend/issues/11327 issue for the on-off switch.

As for the current humidity, it's a more complicated issue. The `humidifier` entity (for which the card is) does not represent the current humidity at all, it is supposed that the current humidity sensor would be linked to the `humidifier` entity via Device Registry. However we cannot use the Device Registry for virtual devices (such as `generic hygrostat`)so I don't know how it should be resolved.

**Reply by sharoltd** [reply]
Oh, I understood. Its sadly... (( Is the situation different for the Generic thermostat? 
And what about icons for On/Off?
![Screenshot_2](https://user-images.githubusercontent.com/48355538/149500608-3a081f9f-4126-4a27-8018-b8577a60cf79.jpg)
![Screenshot_3](https://user-images.githubusercontent.com/48355538/149500758-f015711e-2279-4b62-849f-aca8ef89900b.jpg)

**Reply by Jc2k** [reply]
https://github.com/home-assistant/architecture/discussions/702 seeks a way forward for the virtual entities having a unique id. Maybe that would be a good place to discuss an approach to having virtual devices too.

Other than that, I don't think this meets the criteria for an architecture issue. Typically they are raised by the developer doing the work, otherwise they are just feature requests which the devs typically keep in the forums.

**Reply by sharoltd** [reply]
> I've raised [home-assistant/frontend#11327](https://github.com/home-assistant/frontend/issues/11327) issue for the on-off switch.

Sorry I didn't notice right away. Thanks!

**Comment by sharoltd** [comment]
Also such question. As i see, when HA restarted humidifier swith always turn OFF? Regardless of the conditions that were before the restart. Is it so intended? In the Generic thermostat, as far as I remember, the conditions that were before the restart are restored. Perhaps there is a logic to do so for the humidifier also?

**Reply by Jc2k** [reply]
That again is **not** an architecture issue. AIUI `generic_thermostat` and `generic_hygrostat` are both "restore entities". If you think this is not working you can file a bug report in the proper repo (https://github.com/home-assistant/core/issues/new?assignees=&labels=&template=bug_report.yml).

Make sure that you link to the hygrostat documentation (that is how the bot will connect you with the best developer to help you).

**Reply by sharoltd** [reply]
So it must restore its state? State of the generic_hygrostat or state of hydrostate switch?
Thanks!

**Reply by Shulyaka** [reply]
The state of the `generic_hygrostat` will be restored. The state of the humidifier switch will be set according to the current humidity and generic_hygrostat setpoint.

**Reply by sharoltd** [reply]
Yes, hydrostat restored, confirm. But humidifier switch always turn off at moment when HA restarted (before restsring switch is ON).

**Comment by palves** [comment]
I've setup an hygrostat (a dehumidifier) for the first time today, and I was very surprised to see how its GUI is different from the generic thermostat.  I find it _very_ surprising, whatever the technical reasons behind it.  From a user's perspective, a thermostat and a hygrostat should be no different, apart from using different units (temp vs humidity) and different colors perhaps.  The inconsistency is super puzzling.

I found the discussion in the original github pull request which led me here.

Where did the discussion go to from here?   Did it just fizzle out?
